src/FormControl/FormControl.tsx

```diff
@@ -0,0 +1,232 @@
+          <Box display="flex" alignItems={slots.LeadingVisual ? 'center' : undefined}>
+            <Box sx={{'> input': {marginLeft: 0, marginRight: 0}}}>
+              {React.isValidElement(InputComponent) &&
+                React.cloneElement(InputComponent, {
+                  id,
+                  disabled,
+                  ['aria-describedby']: captionId
+                })}
+              {React.Children.toArray(children).filter(
+                child =>
+                  React.isValidElement(child) &&
+                  ![Checkbox, Radio].some(inputComponent => child.type === inputComponent)
+              )}
+            </Box>
+            {slots.LeadingVisual && (
+              <Box
+                color={disabled ? 'fg.muted' : 'fg.default'}
+                sx={{
+                  '> *': {
+                    minWidth: slots.Caption ? get('fontSizes.4') : get('fontSizes.2'),
+                    minHeight: slots.Caption ? get('fontSizes.4') : get('fontSizes.2'),
+                    fill: 'currentColor'
+                  }
+                }}
+                ml={2}
+              >
+                {slots.LeadingVisual}
+              </Box>
+            )}
+            {(React.isValidElement(slots.Label) && !slots.Label.props.visuallyHidden) || slots.Caption ? (
+              <Box display="flex" flexDirection="column" ml={2}>
+                {slots.Label}
+                {slots.Caption}
+              </Box>
+            ) : (
+              <>
+                {slots.Label}
+                {slots.Caption}
+              </>
+            )}
+          </Box>
+        ) : (
+          <Box
+            display="flex"
+            flexDirection="column"
+            width="100%"
+            sx={isLabelHidden ? {'> *:not(label) + *': {marginTop: 2}} : {'> * + *': {marginTop: 2}}}
+          >
+            {React.Children.toArray(children).filter(
+              child =>
+                React.isValidElement(child) &&
+                !expectedInputComponents.some(inputComponent => child.type === inputComponent)
+            )}
+            {slots.Label}
+            {React.isValidElement(InputComponent) &&
+              React.cloneElement(InputComponent, {
+                id,
+                required,
+                disabled,
+                validationStatus,
+                ['aria-describedby']: [validationMessageId, captionId].filter(Boolean).join(' ')
+              })}
+            {validationChild && (
+              <ValidationAnimationContainer show>
+                <InputValidation validationStatus={validationStatus} id={validationMessageId}>
+                  {slots.Validation}
+                </InputValidation>
+              </ValidationAnimationContainer>
+            )}
+            {slots.Caption}
+          </Box>
+        )
+      }}
+    </Slots>
+  )
+}
+
+FormControl.defaultProps = {
+  variant: 'stack'
+}
+
+export type FormControlComponentProps = ComponentProps<typeof FormControl>
```

Comment:
I don't think `ComponentProps<>` adds much here since this is not a styled-component. Let's just export the props type where it's defined.

```suggestion
```